recipe/patches/0009-Find-ROCm-device-library-in-conda-layout.patch

```diff
@@ -0,0 +1,42 @@
+From 7f79715796023bb7f020c370ba69fc1adc802a33 Mon Sep 17 00:00:00 2001
+From: Isuru Fernando <isuruf@gmail.com>
+Date: Mon, 11 Mar 2024 14:03:12 -0500
+Subject: [PATCH] Find ROCm device library in conda layout
+
+---
+ clang/lib/Driver/ToolChains/AMDGPU.cpp | 19 +++++++++++++++++++
+ 1 file changed, 19 insertions(+)
+
+diff --git a/clang/lib/Driver/ToolChains/AMDGPU.cpp b/clang/lib/Driver/ToolChains/AMDGPU.cpp
+index d0223322b56b..8f461197f880 100644
+--- a/clang/lib/Driver/ToolChains/AMDGPU.cpp
++++ b/clang/lib/Driver/ToolChains/AMDGPU.cpp
+@@ -418,6 +418,25 @@ void RocmInstallationDetector::detectDeviceLibrary() {
+   if (HasDeviceLibrary)
+     return;
+ 
++  // Find device libraries in <LLVM_DIR>/lib/amdgcn/bitcode
++  LibDevicePath = D.getInstalledDir();;
++  llvm::sys::path::append(LibDevicePath, CLANG_INSTALL_LIBDIR_BASENAME,
++                          "amdgcn", "bitcode");
++  HasDeviceLibrary = CheckDeviceLib(LibDevicePath, true);
++  if (HasDeviceLibrary)
++    return;
++
++  // Find device libraries in the real path of the invoked clang, resolving symbolic
++  // link of clang itself.
++  llvm::SmallString<256> RealClangPath;
++  llvm::sys::fs::real_path(D.getClangProgramPath(), RealClangPath);
++  LibDevicePath = llvm::sys::path::parent_path(RealClangPath);
++  llvm::sys::path::append(LibDevicePath, CLANG_INSTALL_LIBDIR_BASENAME,
++                          "amdgcn", "bitcode");
++  HasDeviceLibrary = CheckDeviceLib(LibDevicePath, true);
++  if (HasDeviceLibrary)
++    return;
++
+   // Find device libraries in a legacy ROCm directory structure
+   // ${ROCM_ROOT}/amdgcn/bitcode/*
```

Comment:
Is there a reason why we couldn't/shouldn't just set `ROCM_ROOT=$BUILD_PREFIX/lib` (or wherever relative to where our clang sits)?

Reply:
Yes, because then `ROCM_ROOT/bin` would be `$BUILD_PREFIX/lib/bin`

Reply:
OK, looking at the surrounding [context](https://github.com/llvm/llvm-project/blob/release/17.x/clang/lib/Driver/ToolChains/AMDGPU.cpp#L413-L429), this makes sense; we're not gonna have the rocm-bits in a clang specific path.